### ISSUE TITLE
False positives that I can't seem to make go away

### ISSUE THREAD

**Issue by AlemTuzlak** [webpro/knip#589]
First of all, amazing work on this! 

I've went through everything in the official docs and I'm a bit confused as to why this is not working, I'm hoping to get some insights if I'm doing something wrong or if this is some bug. But I don't think it's the latter. 

So I have this repo:
https://github.com/forge42dev/base-stack

And after running knip on it I get the below report:

![image](https://github.com/webpro/knip/assets/18480956/a44827ad-7802-488e-a638-2c2b4ad26dd8)

So the following deps:
`chalk, node-html-parser prompt` and all the @types are used. 
`remix-development-tools and vite-tsconfig-paths` are both used in vite.config.ts as plugins
`remix-custom-routes` is used inside of remix/config.ts
and all the other ones are used across the projct and yet I get them reported as not used.

I've tried adding the following config:
 ```json
{
  "$schema": "https://unpkg.com/knip@5/schema.json",
  "entry": [
    "app/root.tsx",
    "app/entry.{client,server}.{ts,tsx}",
    "app/**/*.{js,ts,tsx}",
    "scripts/*.{ts,js}",
    "remix/*.{ts,js}",
    "vite.config.ts"
  ],
  "project": [
    "**/*.{js,cjs,mjs,jsx,ts,cts,mts,tsx}",
    "vite.config.{ts}"
  ]
}
```
and then running `npm run knip -- --config knip.json` and `npm run knip` but both don't seem to work. Am I missing something obvious here? I've tried so many different patterns in both entry and project but none seem to work.

**Comment by webpro** [comment]
- You shouldn't need to manually `vite.config.ts` add as entry files.
- Try `--debug` to see what plugins are enabled, and what each plugin returns individually.
- E.g. when the Remix plugin is enabled, the `app/*` entries should be added automatically.

**Comment by AlemTuzlak** [comment]
@webpro I've tried adding these after I had no luck with running it without the config file, here is what debug yields when it comes to plugins:
```json
[
  'ESLint',
  'husky',
  'PostCSS',
  'Remix',
  'Tailwind',
  'TypeScript',
  'Vite',
  'Vitest'
]
```

**Comment by AlemTuzlak** [comment]
deleting the config gives me the same false positives, in theory I just need to add the scripts/*.ts and remix/*.ts to detect those files as well, but I'm a bit confused on the other ones.

**Comment by webpro** [comment]
Oh dang... there's a bug in Knip. If you remove the empty lines in `.gitignore` it all starts to work as expected :)

**Comment by webpro** [comment]
This line isn't empty: https://github.com/forge42dev/base-stack/blob/main/.gitignore#L6

**Comment by AlemTuzlak** [comment]
@webpro ahh you're right! removing the empty lines works, want me to help out with the fix? I'm guessing it's just a matter of skipping empty lines

**Comment by webpro** [comment]
No worries, I have a fix already. Glad we've sorted it out, the next release will skip that whitespace too.

**Comment by AlemTuzlak** [comment]
@webpro awesome, thank you! I'll keep it open so you can reference it with the new release!

For anyone reading this at `5.9.0` just remove empty space in .gitignore!

**Comment by webpro** [comment]
Yeah newlines aren't the issue, but a string containing only a single space character is true-ish in JS, so...

**Comment by AlemTuzlak** [comment]
Yeah I figured out that it was the single space I left in by mistake 😓

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.9.1. See [Release 5.9.1](https://github.com/webpro/knip/releases/tag/5.9.1) for release notes._

_Using Knip in a commercial project? Please consider [sponsoring me](https://github.com/sponsors/webpro)._
